### DISCUSSION TITLE
Support for custom g-codes

### DISCUSSION THREAD

**Discussion by MTrab** [OctoFarm/OctoFarm#250] [Implemented]
A nice feature would be the possibility to add and save custom g-codes

Could be for loading/unloading filaments

**Comment by NotExpectedYet** [comment]
I like it. 

I will keep this in mind for the backend rewrites I'm hopefully starting soon.

**Reply by NotExpectedYet** [reply]
Infact thinking about it. I may load this with the gcodes bulk command above the test input field. 

Then a new UI button to create them. 


I owe people for my buggy last release.

**Comment by NotExpectedYet** [comment]
What would you think to adding variable parsing to this?

**Reply by NotExpectedYet** [reply]
Added and released btw. Do let me know if the Printer Control terminal fixed itself.

**Reply by MTrab** [reply]
Parsing variables could be a nice feature - took me a little while to find the new script feature. But a nice beginning.
Perhaps make description optional.

**Reply by NotExpectedYet** [reply]
I've just pushed an attempted fix to development for enclosure plugin

**Reply by NotExpectedYet** [reply]
Think this one should nab it :)

**Reply by NotExpectedYet** [reply]
Description is now optional on dev as well as editing
